src/shogun/regression/GaussianProcessRegression.cpp

```diff
@@ -76,20 +76,29 @@ bool GaussianProcessRegression::train_machine(std::shared_ptr<Features> data)
 	// check whether given combination of inference method and likelihood
 	// function supports regression
 	require(m_method, "Inference method should not be NULL");
+	if (m_labels)
+	{
+		m_method->set_labels(m_labels);
+	}
+	if (data)
+	{
+		m_method->set_features(data);
+	}
+	if (m_inducing_features)
+	{
+		auto fitc_method = m_method->as<FITCInferenceMethod>();
+		fitc_method->set_inducing_features(m_inducing_features);
+	}
+
 	auto lik=m_method->get_model();
 	require(m_method->supports_regression(), "{} with {} doesn't support "
 			"regression",	m_method->get_name(), lik->get_name());
-	if (data)
+	if (m_seed != -1)
```

Comment:
I am not sure I understand this?

Reply:
if m_seed is initialized, then we need to set the m_seed value to the Inference, maybe should find a better way to represent whether the m_seed are set.

Reply:
But I thought that the seeds are passed on to subobjects automatically no?

Reply:
Do you mean to make `GaussianProcess` derive from `RandomMixin`? now `m_seed` just is defined in class `GaussianProcess`.

Reply:
or you mean to use `add_callback_function` to set the seed of `m_Inference`?

Reply:
Oh I see, yes it would have to be derived from RandomMixing. Could we have GaussianMachine be derived from RandomMixing<Machine>?

Reply:
yes, we can, and then use `add_callback_function` to set the seed of `m_inference`?

Reply:
I think RandomMixin already has that callback for any registered SGObject?

Reply:
oh, i see, thanks :)